x/cheqd/types/did_verification_method_test.go

```diff
@@ -8,290 +8,303 @@ import (
 	"crypto/rand"
 	"crypto/rsa"
 	"encoding/json"
-	"testing"
 
+	. "github.com/cheqd/cheqd-node/x/cheqd/types"
 	"github.com/lestrrat-go/jwx/jwk"
 	"github.com/multiformats/go-multibase"
-	"github.com/stretchr/testify/require"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
 )
 
-type TestJWKKey struct {
-	Kty string `json:"kty"`
-	N   string `json:"n"`
-	Use string `json:"use"`
-	Alg string `json:"alg"`
-	E   string `json:"e"`
-	Kid string `json:"kid"`
-}
-
-var ValidJWKKey = TestJWKKey{
-	Kty: "RSA",
-	N:   "o76AudS2rsCvlz_3D47sFkpuz3NJxgLbXr1cHdmbo9xOMttPMJI97f0rHiSl9stltMi87KIOEEVQWUgMLaWQNaIZThgI1seWDAGRw59AO5sctgM1wPVZYt40fj2Qw4KT7m4RLMsZV1M5NYyXSd1lAAywM4FT25N0RLhkm3u8Hehw2Szj_2lm-rmcbDXzvjeXkodOUszFiOqzqBIS0Bv3c2zj2sytnozaG7aXa14OiUMSwJb4gmBC7I0BjPv5T85CH88VOcFDV51sO9zPJaBQnNBRUWNLh1vQUbkmspIANTzj2sN62cTSoxRhSdnjZQ9E_jraKYEW5oizE9Dtow4EvQ",
-	Use: "sig",
-	Alg: "RS256",
-	E:   "AQAB",
-	Kid: "6a8ba5652a7044121d4fedac8f14d14c54e4895b",
-}
-
-var NotValidJWKKey = TestJWKKey{
-	Kty: "SomeOtherKeyType",
-	N:   "o76AudS2rsCvlz_3D47sFkpuz3NJxgLbXr1cHdmbo9xOMttPMJI97f0rHiSl9stltMi87KIOEEVQWUgMLaWQNaIZThgI1seWDAGRw59AO5sctgM1wPVZYt40fj2Qw4KT7m4RLMsZV1M5NYyXSd1lAAywM4FT25N0RLhkm3u8Hehw2Szj_2lm-rmcbDXzvjeXkodOUszFiOqzqBIS0Bv3c2zj2sytnozaG7aXa14OiUMSwJb4gmBC7I0BjPv5T85CH88VOcFDV51sO9zPJaBQnNBRUWNLh1vQUbkmspIANTzj2sN62cTSoxRhSdnjZQ9E_jraKYEW5oizE9Dtow4EvQ",
-	Use: "sig",
-	Alg: "RS256",
-	E:   "AQAB",
-	Kid: "6a8ba5652a7044121d4fedac8f14d14c54e4895b",
-}
-
-var (
-	ValidJWKByte, _    = json.Marshal(ValidJWKKey)
-	NotValidJWKByte, _ = json.Marshal(NotValidJWKKey)
-)
-
-var (
-	ValidPublicKeyJWK    = JSONToPubKeyJWK(string(ValidJWKByte))
-	NotValidPublicKeyJWK = JSONToPubKeyJWK(string(NotValidJWKByte))
-)
-
-func TestVerificationMethodValidation(t *testing.T) {
-	cases := []struct {
-		name              string
-		struct_           VerificationMethod
-		baseDid           string
-		allowedNamespaces []string
-		isValid           bool
-		errorMsg          string
-	}{
-		{
-			name: "valid method with multibase key",
-			struct_: VerificationMethod{
+var _ = Describe("Verification Method Validation tests", func() {
+	var struct_ VerificationMethod
+	var baseDid string
+	var allowedNamespaces []string
+	var isValid bool
+	var errorMsg string
+
+	BeforeEach(func() {
+		struct_ = VerificationMethod{}
+		baseDid = ""
+		allowedNamespaces = []string{}
+		isValid = false
+		errorMsg = ""
+	})
+
+	AfterEach(func() {
+		err := struct_.Validate(baseDid, allowedNamespaces)
+
+		if isValid {
+			Expect(err).To(BeNil())
+		} else {
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring(errorMsg))
+		}
+	})
+	When("verification method with expected multibase key", func() {
+		It("is valid", func() {
+			struct_ = VerificationMethod{
 				Id:                 "did:cheqd:aaaaaaaaaaaaaaaa#qwe",
```

Comment:
- Doesn't contain namespace
- Use a realistic DID. Check #399 and see the discussion there but it should be 22.

Reply:
* Currently, namespace is mandatory. We can make it optional and restore during requests normalization. Better to do in a separate ticket.
* `Use a realistic DID` Imo tests should be fixed in the corresponding PR (the one that fixes identifier validation).